spec/rubocop/cop/rspec/aggregate_examples_spec.rb

```diff
@@ -0,0 +1,790 @@
+        end
+      end
+    RUBY
+
+    it_behaves_like 'detects and autocorrects', offensive_source, good_source
+  end
+
+  context 'with symbol metadata' do
+    offensive_source = <<-RUBY
+      describe do
+        it { is_expected.to be_so_so }
+        it(:awesome) { is_expected.to be_awesome }
+        it(:awesome, aggregate_failures: true) { is_expected.to be_cool }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above.
+        it(:awesome, :aggregate_failures) { is_expected.to be_amazing }
+      end
+    RUBY
+
+    good_source = <<-RUBY
+      describe do
+        it { is_expected.to be_so_so }
+        specify(:awesome) do
+          is_expected.to be_awesome
+          is_expected.to be_cool
+          is_expected.to be_amazing
+        end
+      end
+    RUBY
+
+    it_behaves_like 'detects and autocorrects', offensive_source, good_source
+  end
+
+  context 'with `aggregate_failures: false` in metadata' do
+    fair_source = <<-RUBY
+      describe do
+        it(:awesome) { is_expected.to be_awesome }
+        it(:awesome, aggregate_failures: false) { is_expected.to be_cool }
+        it(:awesome, aggregate_failures: false) { is_expected.to be_amazing }
+      end
+    RUBY
+
+    it { expect_no_offenses(fair_source) }
+  end
+
+  context 'with mixed aggregate_failures in metadata' do
+    fair_source = <<-RUBY
+      describe do
+        it(:awesome, aggregate_failures: true) { is_expected.to be_cool }
+        it(:awesome, aggregate_failures: false) { is_expected.to be_amazing }
+      end
+    RUBY
+
+    it { expect_no_offenses(fair_source) }
+  end
+
+  context 'with metadata and title' do
+    offensive_source = <<-RUBY
+      describe do
+        it { is_expected.to be_so_so }
+        it(:awesome) { is_expected.to be_awesome }
+        it('is ok', :awesome) { is_expected.to be_ok }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above.
+      end
+    RUBY
+
+    it_behaves_like 'detects, but does not autocorrect', offensive_source
+  end
+
+  context 'with mixed metadata' do
+    offensive_source = <<-RUBY
+      describe do
+        it { is_expected.to be_ok }
+        it(:model, isolation: :full) { is_expected.to be_isolated }
+        it(:model, isolation: :full) { is_expected.to be_save_model }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above.
+      end
+    RUBY
+
+    good_source = <<-RUBY
+      describe do
+        it { is_expected.to be_ok }
+        specify(:model, isolation: :full) do
+          is_expected.to be_isolated
+          is_expected.to be_save_model
+        end
+      end
+    RUBY
+
+    it_behaves_like 'detects and autocorrects', offensive_source, good_source
+  end
+
+  context 'with validation actions with side effects' do
+    context 'without configuration' do
+      let(:cop_config) do
+        { 'MatchersWithSideEffects' => [] }
```

Comment:
How is this different from the cop_config in the outer example group (`let(:cop_config) { {} }`)? Testing “validation actions with side effects” isn’t a special case in my opinion. It’s when we configure the cop with a list of “matchers with side effects” that we need a new example group.

Reply:
This context meant to show the behaviour when the default `MatchersWithSideEffects` option is empty.
See below how a wrong context name disconnected the balancing context.